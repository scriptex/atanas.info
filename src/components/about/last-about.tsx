import * as React from 'react';

import { ExternalLink } from '@components';

export const LastAbout: React.FC = () => (
	<div className="c-article__block">
		<h3>Current occupation</h3>

		<ul>
			<li>
				A Senior Javascript/Typescript engineer and front-end mentor at{' '}
				<strong>
					<ExternalLink href="https://github.com/three11">Three 11</ExternalLink>{' '}
				</strong>
				working on various boutique web applications including XPND, Kinetik Automotive and many more.
			</li>

			<li>
				A Lead front-end engineer at E.ON working on the{' '}
				<strong>
					<ExternalLink href="https://www.red-dot.org/project/eon-home-49256">award winning</ExternalLink>
				</strong>{' '}
				web and mobile smart energy management application E.ON Home.
			</li>

			<li>
				A Senior software developer at{' '}
				<strong>
					<ExternalLink href="https://www.toptal.com/resume/atanas-atanasov">Toptal</ExternalLink>
				</strong>
			</li>

			<li>
				A Founding member and Senior Javascript/Typescript engineer at{' '}
				<strong>
					<ExternalLink href="https://app.usebraintrust.com/talent/782/">Braintrust</ExternalLink>
				</strong>
			</li>

			<li>
				A Senior Javascript/Typescript engineer at{' '}
				<strong>
					<ExternalLink href="https://andela.com/">Andela</ExternalLink>
				</strong>
			</li>

			<li>
				A Senior Frontend engineer at{' '}
				<strong>
					<ExternalLink href="https://app.9am.works/hire/atanas-atanasov">9AM</ExternalLink>
				</strong>
			</li>
		</ul>

		<br />

		<h3>Connect with me</h3>

		<p>
			I am open for hire to work on your awesome idea. If you think I can help you, do not hesitate and{' '}
			<strong>
				<ExternalLink href="mailto:hi@atanas.info">drop me a line</ExternalLink>
			</strong>
			.
		</p>

		<p>
			I also like to contribute to open-source projects and I have created some myself. Feel free to contact me if
			you want to talk about your open-source projects or{' '}
			<strong>
				<ExternalLink href="https://github.com/scriptex">mine</ExternalLink>
			</strong>
			.
		</p>
	</div>
);

export default LastAbout;
