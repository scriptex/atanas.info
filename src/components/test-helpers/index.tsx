import * as React from 'react';
import { act } from 'react-dom/test-utils';
import { render, RenderResult, waitFor } from '@testing-library/react';

import * as shared from '../../scripts/shared';

export const mockUseEffect = (): void => {
	let useEffect: jest.Mock<React.EffectCallback>;

	const mockedUseEffect = () => {
		useEffect.mockImplementationOnce((f: () => React.EffectCallback) => f());
	};

	beforeEach(() => {
		useEffect = jest.spyOn(React, 'useEffect') as unknown as jest.Mock<React.EffectCallback>;

		mockedUseEffect();
	});
};

export const snapshotTest = (
	Component: React.FC<Readonly<React.ComponentProps<any>>>,
	buttonSelector?: string
): void => {
	const { name } = Component;

	jest.spyOn(shared, 'formatDate');

	(shared.formatDate as jest.Mock).mockImplementation(() => 'Mock date');

	mockUseEffect();

	afterAll(() => {
		jest.clearAllMocks();
	});

	describe(`${name} component`, () => {
		it(`Should render the ${name} component`, async () => {
			let result!: RenderResult;

			await act(async () => {
				result = await waitFor(() => render(<Component />));
			});

			const { asFragment, container } = result;

			expect(asFragment()).toMatchSnapshot();

			if (buttonSelector) {
				act(() => {
					container.querySelectorAll<HTMLElement>(buttonSelector).forEach(button => {
						if (!button.hasAttribute('download')) {
							button.click();

							expect(asFragment()).toMatchSnapshot();
						}
					});
				});
			}
		});

		it(`Should unmount the ${name} component`, async () => {
			let result!: RenderResult;

			await act(async () => {
				result = await waitFor(() => render(<Component />));
			});

			const { unmount, asFragment } = result;

			unmount();

			expect(asFragment()).toMatchSnapshot();
		});
	});
};
