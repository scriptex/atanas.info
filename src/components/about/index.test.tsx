import { About } from '.';
import { LastAbout } from './last-about';
import { snapshotTest } from '../test-helpers';

snapshotTest(About);
snapshotTest(LastAbout);
