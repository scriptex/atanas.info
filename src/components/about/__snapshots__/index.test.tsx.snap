// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Should render the About component 1`] = `
<DocumentFragment>
  <main
    class="o-main "
  >
    <section
      class="c-section c-section--about c-article fullsize-background"
      id="about"
      style="background-image: url(/images/temp/desktop.jpg);"
    >
      <div
        class="o-shell"
      >
        <header
          class="c-section__header"
        >
          <div
            class="c-section__animation"
            style="width: 200px; height: 200px; margin-left: auto; margin-right: auto;"
          />
          <h2>
            About me
          </h2>
        </header>
        <img
          alt="Atanas Atanasov smiling dressed in a green t-shirt"
          height="240"
          loading="lazy"
          src="/images/temp/atanas.jpg"
          width="240"
        />
        <div
          class="c-article__block"
        >
          <h3>
            Featured
          </h3>
          <span
            class="c-article__block-emoji"
          >
            ðŸŽ‰ ðŸŽ‰ ðŸŽ‰
          </span>
          <p>
            My 
            <a
              href="https://scriptex.tk/"
              rel="noopener noreferrer"
              target="_blank"
            >
              interactive resume
            </a>
            , which is built using the 
            <a
              href="https://docs.codersrank.io/widgets/"
              rel="noopener noreferrer"
              target="_blank"
            >
              CodersRank widgets
            </a>
            , is featured in the 
            <a
              href="https://blog.codersrank.io/10-inspiring-developer-websites-and-profile-pages/"
              rel="noopener noreferrer"
              target="_blank"
            >
              CodersRank blog
            </a>
             !!!
          </p>
          <p>
            My 
            <a
              href="https://profile.codersrank.io/user/scriptex/"
              rel="noopener noreferrer"
              target="_blank"
            >
              CodersRank profile
            </a>
             is featured in the 
            <a
              href="https://codersrank.io/for-companies/"
              rel="noopener noreferrer"
              target="_blank"
            >
              CodersRank candidate platform
            </a>
            !!!
            <br />
            Feel free to check 
            <a
              href="https://codersrank.io/wp-content/uploads/2020/09/CR-search.mp4"
              rel="noopener noreferrer"
              target="_blank"
            >
              the video
            </a>
             out!
          </p>
          <span
            class="c-article__block-emoji"
          >
            ðŸŽ‰ ðŸŽ‰ ðŸŽ‰
          </span>
        </div>
        <div
          class="c-article__block"
        >
          <h3>
            2010
          </h3>
          <p>
            My name is Atanas and I am a Senior JavaScript/TypeScript engineer.
          </p>
          <p>
            I am currently located in Varna, Bulgaria. That's on the south-east side of Europe, on the Black Sea.
          </p>
          <p>
            I started my web development career in 2010 as a freelance apprentice to a hardcore fullstack web developer and pretty fast learned the basics of HTML and CSS.
          </p>
          <p>
            We even had our own small web development company. In "HTML Lab" I was doing the front-end part and was mainly focused on slicing design files and turning them into static HTML pages.
          </p>
        </div>
        <div
          class="c-article__block"
        >
          <h3>
            2012
          </h3>
          <p>
            Between 2012 and 2018 I was employed in one of the largest web development companies in Varna, Bulgaria - 2create. In this company I worked for some of the biggest PSD-to-HTML brands out there - htmlBurger and htmlBoutique. I developed thousands (yes thousands) of different projects for clients from all over the globe and was focused on the front-end part of all projects.
          </p>
          <p>
            In the last 3 years of my employment there I was a direct assistant to several project managers and I also mentored several junior and intermediate front-end developers.
          </p>
          <p>
            During the period I participated in defining, integrating and teaching the internal coding standards in the company and co-authored the internal front-end projects starter template which was based on Webpack, Babel, PostCSS, SCSS and many more. The coding standards and Jarvis - that was the name of the front-end template - were (and are) used in tens of thousands of projects for clients from around the world.
          </p>
        </div>
        <div
          class="c-article__block"
        >
          <h3>
            2014
          </h3>
          <p>
            During my employment between 2012 and 2018 I developed, integrated and maintained the internal build systems for the company. This included mastering CSS pre- and post-processors such as LESS, SASS/SCSS and PostCSS, task runners such as Gulp and Grunt, module bundlers such as Webpack and Rollup, and NodeJS which I used as a runtime.
          </p>
          <p>
            Along these technologies I mastered JavaScript and it became my preferred coding language. I also presented a keynote on the features introduced in 2015 with the 6th version of ECMAScript (the JavaScript standard).
          </p>
          <p>
            In 2017 I started learning TypeScript which later became the number one tool in my utilities belt. I prefer starting new projects entirely in TypeScript regardless of the targeted platform or end - Web/Native or front-end/back-end.
          </p>
        </div>
        <div
          class="c-article__block"
        >
          <h3>
            2018
          </h3>
          <p>
            Since the beginning of 2018 I am working as a JavaScript/TypeScript engineer in a small startup located in Varna and operating internationally.
          </p>
          <p>
            At Three 11 we are focused on providing the best possible quality while using bleeding edge technologies. We are working on several in-house projects along with some interesting client websites and applications.
          </p>
          <p>
            We prefer JavaScript and TypeScript and utilize their power when building applications using React, Angular, Vue, React Native and even static web sites or WordPress powered CMS websites and blogs.
          </p>
        </div>
        <div
          class="c-article__block"
        >
          <h3>
            Today
          </h3>
          <p>
            Now I am a JavaScript and TypeScript engineer with more than 10 years of professional experience. I am focused on the front-end but using a lot of back-end technologies as well. My daily tasks include coding HTML, CSS, JavaScript, TypeScript, PHP, Bash and many more.
          </p>
          <p>
            The latest project I worked on is an award-winning, large, international IoT application for one of the world's largest investor-owned electric utility service providers which aims to provide users from all over Europe with the ability to control their homes' smart appliances and energy consumption.
          </p>
        </div>
        <div
          class="c-article__block"
        >
          <h3>
            Current occupation
          </h3>
          <ul>
            <li>
              A Senior JavaScript/TypeScript engineer and front-end mentor at 
              <strong>
                <a
                  href="https://github.com/three11"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Three 11
                </a>
              </strong>
              working on various boutique web applications including XPND, Kinetik Automotive and many more.
            </li>
            <li>
              A Lead front-end developer at E.ON working on the 
              <strong>
                <a
                  href="https://www.red-dot.org/project/eon-home-49256"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  award winning
                </a>
              </strong>
               web and mobile smart energy management application E.ON Home.
            </li>
            <li>
              A Senior software developer at 
              <strong>
                <a
                  href="https://www.toptal.com/resume/atanas-atanasov"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Toptal
                </a>
              </strong>
            </li>
            <li>
              A Founding member and Senior JavaScript/TypeScript engineer at 
              <strong>
                <a
                  href="https://app.usebraintrust.com/talent/782/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Braintrust
                </a>
              </strong>
            </li>
          </ul>
          <br />
          <h3>
            Connect with me
          </h3>
          <p>
            I am open for hire to work on your awesome idea. If you think I can help you, do not hesitate and 
            <strong>
              <a
                href="mailto:hi@atanas.info"
                rel="noopener noreferrer"
                target="_blank"
              >
                drop me a line
              </a>
            </strong>
            .
          </p>
          <p>
            I also like to contribute to open-source projects and I have created some myself. Feel free to contact me if you want to talk about your open-source projects or 
            <strong>
              <a
                href="https://github.com/scriptex"
                rel="noopener noreferrer"
                target="_blank"
              >
                mine
              </a>
            </strong>
            .
          </p>
        </div>
        <div
          class="c-section__actions"
        >
          <button
            class="c-btn "
            type="button"
          >
            Get in touch
          </button>
          <ul
            class="c-section__actions-list"
          >
            <li>
              <button
                class="c-btn c-section__actions-close"
                type="button"
              >
                Close
              </button>
            </li>
            <li>
              <button
                class="c-btn "
                type="button"
              >
                <svg
                  class="c-section__actions-icon"
                >
                  <use
                    xlink:href="#svg-email"
                  />
                </svg>
                <span>
                  Contact me
                </span>
                <strong>
                  directly
                </strong>
              </button>
            </li>
            <li>
              <a
                class="c-btn"
                href="https://app.usebraintrust.com/talent/782/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="c-section__actions-icon"
                >
                  <use
                    xlink:href="#svg-braintrust"
                  />
                </svg>
                <span>
                  Hire me on
                </span>
                <strong>
                  Braintrust
                </strong>
              </a>
            </li>
            <li>
              <a
                class="c-btn"
                href="https://www.toptal.com/resume/atanas-atanasov"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="c-section__actions-icon"
                >
                  <use
                    xlink:href="#svg-toptal"
                  />
                </svg>
                <span>
                  Hire me on
                </span>
                <strong>
                  Toptal
                </strong>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;

exports[`LastAbout component Should render the LastAbout component 1`] = `
<DocumentFragment>
  <div
    class="c-article__block"
  >
    <h3>
      Current occupation
    </h3>
    <ul>
      <li>
        A Senior JavaScript/TypeScript engineer and front-end mentor at 
        <strong>
          <a
            href="https://github.com/three11"
            rel="noopener noreferrer"
            target="_blank"
          >
            Three 11
          </a>
        </strong>
        working on various boutique web applications including XPND, Kinetik Automotive and many more.
      </li>
      <li>
        A Lead front-end developer at E.ON working on the 
        <strong>
          <a
            href="https://www.red-dot.org/project/eon-home-49256"
            rel="noopener noreferrer"
            target="_blank"
          >
            award winning
          </a>
        </strong>
         web and mobile smart energy management application E.ON Home.
      </li>
      <li>
        A Senior software developer at 
        <strong>
          <a
            href="https://www.toptal.com/resume/atanas-atanasov"
            rel="noopener noreferrer"
            target="_blank"
          >
            Toptal
          </a>
        </strong>
      </li>
      <li>
        A Founding member and Senior JavaScript/TypeScript engineer at 
        <strong>
          <a
            href="https://app.usebraintrust.com/talent/782/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Braintrust
          </a>
        </strong>
      </li>
    </ul>
    <br />
    <h3>
      Connect with me
    </h3>
    <p>
      I am open for hire to work on your awesome idea. If you think I can help you, do not hesitate and 
      <strong>
        <a
          href="mailto:hi@atanas.info"
          rel="noopener noreferrer"
          target="_blank"
        >
          drop me a line
        </a>
      </strong>
      .
    </p>
    <p>
      I also like to contribute to open-source projects and I have created some myself. Feel free to contact me if you want to talk about your open-source projects or 
      <strong>
        <a
          href="https://github.com/scriptex"
          rel="noopener noreferrer"
          target="_blank"
        >
          mine
        </a>
      </strong>
      .
    </p>
  </div>
</DocumentFragment>
`;
